### PR TITLE
Custom scroll in year picker and allow editing button

### PR DESCRIPTION
**Pull request by TVGSoft**
Enhance some features:

1. Allow rendering custom scroll component in YearPicker that prevent DatePickerModal inside FlatList.
2. Custom "allowEditing" props to enable or disable the "Edit" button via DateInput.

Fixed bug from #201

### CODE REVIEW

**Comment by RichardLindhout** [comment]
@TVGSoft looks good but we let it slip, could you fix the merge conflict on this so we could merge?

**Comment by TVGSoft** [comment]
Hi @RichardLindhout 

I have resolved the conflict, please help to review and merge this PR.

Thank a lot!

File: src/Date/DatePickerModalContentHeader.tsx [hunk 1]
```diff
@@ -62,19 +63,18 @@ export default function DatePickerModalContentHeader(
...
   const label = getLabel(props.locale, props.mode, props.label)
 
   const color = useHeaderTextColor()
-  const supportingTextColor = theme.isV3 ? theme.colors.onSurfaceVariant : color
-  let allowEditing = mode !== 'multiple'
 
-  if (inputDate !== undefined) {
-    allowEditing = inputDate
-  }
+  const isAllowEditing = allowEditing && mode !== 'multiple' && inputDate !== undefined
```
Comment (jcloquell) [inline src/Date/DatePickerModalContentHeader.tsx:73]: I think this change brought some undesired side effects, as
```
if (inputDate !== undefined) {
    allowEditing = inputDate
}
```
is not equivalent to making `const isAllowEditing = ... && inputDate !== undefined`. 

The first is overriding the value of `isAllowEditing` with `inputDate` if this is defined.
The second is having an impact on the `isAllowEditing` value depending on `inputDate` being defined or not.

At the same time, the existing `inputDate` property is and was never set from anywhere outside where this component is used, so its value is always `undefined`.

All of this make the edit button to always be hidden regardless the value passed for the `allowEditing` prop.

**Review by jcloquell — COMMENTED** [review]
Hi there! I know this was already merged (and also released in the 0.14.0 version), but I found these issues that I wanted to share with you, in case you want to address them :)

File: src/Date/DatePickerModalContentHeader.tsx (hunk 1 above)
Comment (iM-GeeKy) [reply src/Date/DatePickerModalContentHeader.tsx:73]: Hi @jcloquell, thanks for catching this! Would you mind creating a quick PR to address this?

Comment (TVGSoft) [reply src/Date/DatePickerModalContentHeader.tsx:73]: @jcloquell i think can change to this:
`const isAllowEditing = allowEditing && mode !== 'multiple'`

And remove this:
`if (inputDate !== undefined) {
    allowEditing = inputDate
}`

What do you think?

Comment (jcloquell) [reply src/Date/DatePickerModalContentHeader.tsx:73]: Sure! Here is the PR - https://github.com/web-ridge/react-native-paper-dates/pull/264 :)

**Pull request merged**